### PR TITLE
build(ci): Trigger getsentry backend tests on deps change

### PR DESCRIPTION
**Pull request by billyvg**
This will trigger getsentry backend test when `requirements*.txt`
changes. This functions similarly to the getsentry javascript test.

The failed check will show you a message w/ instructions on how to proceed:
image

### CODE REVIEW

File: .github/file-filters.yml [hunk 1]
```diff
@@ -29,13 +29,16 @@ frontend_components_modified_lintable:
 backend_lintable: &backend_lintable
   - '**/*.py'
 
+backend_dependencies: &backend_dependencies
+  - 'requirements-*.txt'
```
Comment (billyvg) [inline .github/file-filters.yml:33]: This is what we'll use to determine if a dispatch to getsentry is required.

File: .github/workflows/getsentry-dispatch.yml [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+name: getsentry dispatcher
+
+on:
+  # XXX: We are using `pull_request_target` instead of `pull_request` because we want
+  # this to run on forks.  It allows forks to access secrets safely by
+  # only running workflows from the main branch. Prefer to use `pull_request` when possible.
+  #
+  # See https://github.com/getsentry/sentry/pull/21600 for more details
+  # TODO: this needs to be pull_request_target
+  pull_request:
```
Comment (billyvg) [inline .github/workflows/getsentry-dispatch.yml:11]: this is `pull_request` for testing

File: .github/workflows/scripts/getsentry-dispatch.js [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+      frontend: fileChanges.frontend !== 'true',
+      backend_dependencies: fileChanges.backend_dependencies !== 'true',
+    };
+
+    DISPATCHES.forEach(({workflow, pathFilterName}) => {
+      github.actions.createWorkflowDispatch({
+        owner: 'getsentry',
+        repo: 'getsentry',
+        workflow_id: workflow,
+        ref: 'build/ci/add-backend-dependencies-test', // TODO: this needs to be 'master'
```
Comment (billyvg) [inline .github/workflows/scripts/getsentry-dispatch.js:34]: This is for testing only.

**Comment by billyvg** [comment]
You can view it working here: https://github.com/getsentry/getsentry/runs/1859931351?check_suite_focus=true

(And javascript still works as expected as well https://github.com/getsentry/getsentry/actions/runs/549994891)

**Review by evanpurkhiser — APPROVED** [review]

File: .github/file-filters.yml (hunk 1 above)
Comment (joshuarli) [reply .github/file-filters.yml:33]: You should probably add that as a comment.

**Review by joshuarli — APPROVED** [review]

**Pull request merged**